Add docker support

ga-lep opened the pull request.

Commit:
Adds docker support

Commit:
Improve Docker support

Rewrites Installation part of README.md
Restructures Dockerfile
Adds docker push github workflow

Septimus4 commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,43 @@
+      - name: Log in to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
I will look to add them

Septimus4 commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,43 @@
+on:
+  push:
+    branches: ["main"] # Change to your default branch
+    tags: ["v*"] # Trigger on version tags (e.g., v1.0.0)
```
Can you update to only publish on tags, remove push to main.

Septimus4 commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,43 @@
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```
GHCR requires lowercase names `${{ toLower(github.repository) }}`

Septimus4 commented on README.md:
````diff
@@ -23,6 +26,32 @@ For development extras (linting + tests):
+then use it:
+
+```bash
+docker run ghcr.io/Septimus4/Chronogen [ARGS]
````
Needs lowercase and a tag (either :latest or :${VERSION}), for the readme let's go with latest

Septimus4 commented on .github/workflows/docker-publish.yml:
```diff
@@ -0,0 +1,43 @@
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ steps.version.outputs.version }}
```
can you add OCI labels
and add caching to speed up next builds
```
with:
  cache-from: type=gha
  cache-to: type=gha,mode=max
```

Septimus4 reviewed (changes requested):
Thanks for the work

Commit:
Changes docker workflow

Updates docker workflow to handle lowercase naming and OCI labels
Updates README.md to lowercase docker label and adds tag
